### PR TITLE
Fix /_error not rendering 404 in development

### PR DESCRIPTION
**Pull request by ijjk**
In development if you visit `/_error` the error page is rendered incorrectly and instead should render a 404 since it is a blocked page along with `_app` and `_document`.

### CODE REVIEW

**Review by Timer — CHANGES_REQUESTED** [review]
This doesn't seem right -- how do you edit/develop your error component?

We need to figure out why it doesn't render when visiting it.

File: packages/next-server/lib/constants.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const SERVER_DIRECTORY = 'server'
 export const SERVERLESS_DIRECTORY = 'serverless'
 export const CONFIG_FILE = 'next.config.js'
 export const BUILD_ID_FILE = 'BUILD_ID'
-export const BLOCKED_PAGES = ['/_document', '/_app']
+export const BLOCKED_PAGES = ['/_document', '/_app', '/_error']
```
Comment (timneutkens) [inline packages/next-server/lib/constants.ts]: This is also used during `next build` with serverless target I think 🤔

**Comment by timneutkens** [comment]
> We need to figure out why it doesn't render when visiting it.

Same feedback from me, should figure out why it doesn't finish compiling, probably related to on-demand-entries.

**Comment by Timer** [comment]
Closing this stale PR in favor of the issue: https://github.com/zeit/next.js/issues/8036

**Pull request closed without merging**